extensions/cdc-core/src/main/java/com/hazelcast/jet/cdc/impl/HazelcastListDatabaseHistory.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.cdc.impl;
+
+import com.hazelcast.collection.IList;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.instance.impl.HazelcastInstanceFactory;
+import com.hazelcast.internal.util.Preconditions;
+import io.debezium.config.Configuration;
+import io.debezium.document.Document;
+import io.debezium.document.DocumentReader;
+import io.debezium.document.DocumentWriter;
+import io.debezium.relational.history.AbstractDatabaseHistory;
+import io.debezium.relational.history.DatabaseHistoryException;
+import io.debezium.relational.history.DatabaseHistoryListener;
+import io.debezium.relational.history.HistoryRecord;
+import io.debezium.relational.history.HistoryRecordComparator;
+
+import java.io.IOException;
+import java.util.function.Consumer;
+
+/**
+ * Database history implementation backed by Hazelcast IList {@link IList}.
+ *
+ * @since 4.1
+ */
+public class HazelcastListDatabaseHistory extends AbstractDatabaseHistory {
```

Comment:
how do these lists get cleaned up?

Reply:
No idea, this was code I inherited and never thought of checking. Will do so now.

Reply:
Fixed clean-up in it (has a `stop()` method, which is called by the connector and the list will be destroyed there).